packages/grafana-ui/src/components/Table/models.ts

```diff
@@ -0,0 +1,72 @@
+import { ComponentType } from 'react';
+import { DataFrame, Field, formattedValueToString, GrafanaTheme, getDisplayProcessor } from '@grafana/data';
+
+export interface TableColumn {
+  // React table props
+  Header: string;
+  accessor: string | Function;
+  Cell?: (props: ReactTableCellProps) => string | ComponentType<ReactTableCellProps>;
+  // Grafana additions
+  field: Field;
+}
+
+export interface TableRow {
+  [x: string]: any;
+}
+
+export function getTableRows(data: DataFrame): TableRow[] {
+  const tableData = [];
+
+  for (let i = 0; i < data.length; i++) {
+    const row: { [key: string]: string | number } = {};
+    for (let j = 0; j < data.fields.length; j++) {
+      const prop = data.fields[j].name;
+      row[prop] = data.fields[j].values.get(i);
+    }
+    tableData.push(row);
+  }
+
+  return tableData;
+}
+
+export function getColumns(data: DataFrame, theme: GrafanaTheme): TableColumn[] {
+  const cols: TableColumn[] = [];
+
+  for (const field of data.fields) {
+    // Add display processor if there is non already
+    if (!field.display) {
+      field.display = getDisplayProcessor({
```

Comment:
this makes me nervous...  if you call 
```
applyFieldOverrides(options: GetFieldDisplayValuesOptions): DataFrame[] 
```
It will have a processor applied to all fields at the end

Reply:
ok, but that would be outside the visualization I think, so table component itself would not do any of this. Can use applyFieldOverrides in tests & storybook